tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+        assert isinstance(layer['queue'][2], qml.CNOT)
+        assert isinstance(layer['queue'][3], qml.CNOT)
+        assert isinstance(layer['observable'][0], qml.PauliX)
+        assert isinstance(layer['observable'][1], qml.PauliY)
+        assert isinstance(layer['observable'][2], qml.PauliZ)
+
+        # fourth layer subcircuit
+        layer = res[(5, 6, 7)]
+        assert len(layer['queue']) == 9
+        assert len(layer['observable']) == 3
+        assert isinstance(layer['observable'][0], qml.PauliX)
+        assert isinstance(layer['observable'][1], qml.PauliY)
+        assert isinstance(layer['observable'][2], qml.PauliZ)
+
+    def test_evaluate_subcircuits(self, tol):
+        """Test subcircuits evaluate correctly"""
+        dev = qml.device('default.qubit', wires=2)
+
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        # construct subcircuits
+        circuit = qml.QNode(circuit, dev)
+        circuit.construct_metric_tensor([1, 1, 1])
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate subcircuits
+        circuit.metric_tensor(a, b, c)
+
+        # first parameter subcircuit
+        assert np.allclose(circuit._metric_tensor_subcircuits[(0,)]['result'], 0.25, atol=tol, rtol=0)
+
+        # second parameter subcircuit
+        res = circuit._metric_tensor_subcircuits[(1,)]['result']
+        expected = np.cos(a)**2/4
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        # third parameter subcircuit
+        res = circuit._metric_tensor_subcircuits[(2,)]['result']
+        expected = (3-2*np.cos(a)**2*np.cos(2*b)-np.cos(2*a))/16
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_evaluate_diag_metric_tensor(self, tol):
+        """Test that a diagonal metric tensor evaluates correctly"""
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate metric tensor
+        g = circuit.metric_tensor(a, b, c)
+
+        # check that the metric tensor is correct
+        expected = np.array([1, np.cos(a)**2, (3-2*np.cos(a)**2*np.cos(2*b)-np.cos(2*a))/4])/4
+        assert np.allclose(g, np.diag(expected), atol=tol, rtol=0)
+
+    def test_evaluate_block_diag_metric_tensor(self, tol):
+        """Test that a block diagonal metric tensor evaluates correctly,
+        by comparing it to a known analytic result as well as numerical
+        computation."""
+        dev = qml.device('default.qubit', wires=3)
+
+        def non_parametrized_layer(a, b, c):
```

Comment:
Name is misleading; seems to be some parameters in here?

Reply:
Non-parametrized in that this layer has fixed values when used within the QNodes. E.g., a, b, c are defined _outside_ the QNode, and are not considered free parameters of the circuit